backend/addresses/serializers.py

```diff
@@ -0,0 +1,91 @@
+        return value
+
+    def validate_city(self, value):
+        if not value.strip():
+            raise serializers.ValidationError("City is required.")
+        return value
+
+    def validate_governorate(self, value):
+        if value not in EGYPT_GOVERNORATES:
+            raise serializers.ValidationError("Invalid governorate.")
+        return value
+
+    def validate_country(self, value):
+        if value.strip().lower() != "egypt":
+            raise serializers.ValidationError("Country must be 'Egypt'.")
+        return value
+
+    def validate(self, attrs):
+        user = self.context["request"].user
+        is_default = attrs.get("is_default", False)
+
+        if is_default and user.role != "customer":
+            raise serializers.ValidationError("Only customers can set a default shipping address.")
+        return attrs
+
+    def create(self, validated_data):
+        user = self.context['request'].user
+        is_default = validated_data.pop("is_default", False)
+
+        address = Address.objects.create(user=user, **validated_data)
+        if user.role == "customer":
+            if is_default:
+                address.is_default = True
+                address.save(update_fields=["is_default"])
+
+                try:
+                    profile = user.customerprofile
+                    profile.default_shipping_address = address
+                    profile.save(update_fields=["default_shipping_address"])
+                except CustomerProfile.DoesNotExist:
+                    raise serializers.ValidationError("Customer profile must be created before assigning a default address.")
+
+            return address
+
+    def update(self, instance, validated_data):
+        is_being_defaulted = validated_data.get("is_default", instance.is_default)
+        updated_address = super().update(instance, validated_data)
+        if updated_address.user.role == "customer":
+            if is_being_defaulted:
+                updated_address.is_default = True
+                updated_address.save(update_fields=["is_default"])
+
+                try:
+                    profile = updated_address.user.customerprofile
+                    profile.default_shipping_address = updated_address
+                    profile.save(update_fields=["default_shipping_address"])
+                except CustomerProfile.DoesNotExist:
+                    raise serializers.ValidationError("Customer profile must be created before assigning a default address.")
+            else:
+                try:
+                    profile = updated_address.user.customerprofile
+                    if profile.default_shipping_address == updated_address:
+                        profile.default_shipping_address = None
+                        profile.save(update_fields=["default_shipping_address"])
+                except CustomerProfile.DoesNotExist:
+                    pass
```

Comment:
This section shall not fail silently; please raise a `ValidationError` when the `CustomerProfile.DoesNotExist` error isn't raised.
```suggestion
                except CustomerProfile.DoesNotExist:
                    raise ValidationError("Customer profile must exist to remove default address.")
```